server/src/server.ts

```diff
@@ -69,6 +69,9 @@ const validateTextDocumentOnEvent = (event: TextDocumentChangeEvent) =>
   validateTextDocument(event.document);
 documents.onDidOpen(validateTextDocumentOnEvent);
 documents.onDidSave(validateTextDocumentOnEvent);
+documents.onDidClose((event: TextDocumentChangeEvent) =>
+  connection.sendDiagnostics({ uri: event.document.uri, diagnostics: [] }),
+);

```

Comment:
What's this for? It does not seem entirely related to the rest of the PR.

Reply:
Yeah, it should be in another PR / commit.
This is to have the diagnostics for a closed file removed from the UI, to avoid pollution.
Can I keep it in this PR, or should I do another one?

Reply:
Should be OK, but a comment like…

```
// Clearing diagnostic for the closed file to avoid spamming the user.
```

… would be nice.

Reply:
After this, I should be able to approve.